scratch.Rmd

```diff
@@ -0,0 +1,556 @@
+                      #Depth_to_Water_ft = contains('Level..Depth.to.Water..ft.')) %>%
+  mutate(DT = ymd_hms(DT)) %>%
+  mutate(DT = DT - lubridate::hours(1)) %>% # reported in MDT >:(
+  mutate(DT = as.character(round_date(ymd_hms(DT), "15 minutes"))) %>%
+  mutate(DT = ymd_hms(DT))
+
+# Downloading VuLink data:
+raw_tube_legacy <- map_dfr(grep(list.files("data/sensor_data/2022/legacy/", full.names = T), pattern = "VuLink", invert = F, value = T, ignore.case = F), tube_reader) %>%
+  select(DT = 1,
+         PVC_Temp = 2,
+         Battery = 3,
+         Air_Baro = 4) %>%
+  mutate(DT = ymd_hms(DT)) %>%
+  mutate(DT = DT - lubridate::hours(7)) %>%
+  mutate(DT = as.character(round_date(ymd_hms(DT), "15 minutes"))) %>%
+  mutate(DT = ymd_hms(DT))  %>%
+  mutate_at(vars(2:ncol(.)), as.numeric)
+
+raw_legacy <- map_dfr(grep(list.files("data/sensor_data/2022/legacy/", full.names = T), pattern = "TROLL", invert = F, value = T, ignore.case = F), hydrovu_reader)
+names(raw_legacy) <- make.names(names(raw_legacy), unique = T)
+
+rawless_legacy <- raw_legacy %>% 
+select(DT = contains('Date.Time'),
+                      Water_Temp_C = as.numeric(contains('Temperature..C')),
+                      pH = contains('pH'),
+                      ORP_mV = contains('ORP'),
+                      #Actual_Conductivity_µS_cm = contains('Actual.Conductivity..µS.cm.'),
+                      Specific_Conductivity_µS_cm = contains('Specific.Conductivity..µS.cm.'),
+                      #Salinity_PSU = contains('Salinity..psu.'),
+                      DO_ppm = contains('DO..mg'),
+                      Turbidity_NTU = contains('Turbidity'),
+                      #TSS_mg_L = contains('Total.Suspended.Solids..mg.L.'),
+                      #Chla_RFU = contains('Chl.a.Fluorescence..RFU.'),
+                      #Chla_µg_L = contains('Chl.a.Concentration..µg.L.'),
+                      #Pressure_PSI = contains('Pressure..psi.'),
+                      Depth_ft = contains('Depth..ft')) %>%
+                      #Elevation_m = contains('Level..Elevation..m.'),
+                      #Depth_to_Water_ft = contains('Level..Depth.to.Water..ft.')) %>%
+  mutate(DT = ymd_hms(DT)) %>%
+  mutate(DT = DT - lubridate::hours(7)) %>%
+  rbind(rawless_troll_legacy) %>%
+  mutate_at(vars(2:ncol(.)), as.numeric) %>%
+  mutate(site = "legacy") %>%
+  arrange(ymd_hms(DT)) %>%
+  mutate(DT = as.character(round_date(ymd_hms(DT), "15 minutes"))) %>%
+  mutate(DT = ymd_hms(DT)) %>%
+  full_join(raw_tube_legacy, by='DT') %>%
+  dplyr::bind_rows(., filter(pre_2022, site == "legacy")) %>%
+  full_join(filter(field_notes, site == "legacy"), by = c('DT','site')) %>%
+  mutate(date = as_date((DT)),
+         hour = hour(DT),
+         year = year(DT)) %>%
+  arrange(ymd_hms(DT)) %>%
+   # remove data ranges where sensor was pulled out of field (THAT I KNOW OF!).
+  dplyr::filter(!ymd_hms(DT) >= ymd_hms('2021-12-04 19:30:00') & ymd_hms(DT) < ymd_hms('2022-04-06 17:30:00'),
```

Comment:
instead of filter, consider using a function (over your list of dates where you have visits listed) where you recode data to NA instead of dropping observations. An example where I use a loop is here (https://github.com/Lake-Sunapee-Protective-Association/LSPA_buoy/blob/master/Sunapee_buoy2_2021.R) starting at line 182.